### PR TITLE
UNR 2168: Remove QBI toggle

### PR DESCRIPTION
**Pull request by sjtsilvester**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Remove the ability to disable QBI from within the editor and removing streaming queries from simulated player worker configs.

Other PRs for this ticket:
https://github.com/spatialos/UnrealGDKExampleProject/pull/59
https://github.com/improbableio/UnrealEngine/pull/262
https://github.com/spatialos/UnrealGDKThirdPersonShooter/pull/95

#### Release note
Chunk based interest is no longer supported. All interest is resolved through QBI. You should remove streaming query and chunk based interest options from worker and launch config files to avoid unnecessary streaming queries being generated.

#### Tests
Ran example project with simplayers and verified that no streaming queries were still generated.

#### Documentation
Breaking change release note.

#### Primary reviewers
@m-samiec @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 1]
```diff
@@ -147,11 +147,6 @@ Interest InterestFactory::CreateServerWorkerInterest()
 
 Interest InterestFactory::CreateInterest() const
 {
-	if (!GetDefault<USpatialGDKSettings>()->bUsingQBI)
-	{
-		return Interest{};
-	}
-
 	if (GetDefault<USpatialGDKSettings>()->bEnableServerQBI)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: What happens when `bEnableServerQBI` is false? Does it just check out everything?

Comment (sjtsilvester) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: Yep (each entity gets an interest query with component constraint for the spatial position)

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: Oh, I think you mean the special server entity has an interest query with a component constraint of spatial position. I think I get what's going on there now.

Comment (sjtsilvester) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: That is indeed what I meant. (With server QBI on, each entity gets an interest query)

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 2]
```diff
@@ -418,43 +418,6 @@ bool FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig() const
 	const USpatialGDKSettings* SpatialGDKRuntimeSettings = GetDefault<USpatialGDKSettings>();
 	const FSpatialLaunchConfigDescription& LaunchConfigDescription = SpatialGDKEditorSettings->LaunchConfigDesc;
 
-	if (const FString* EnableChunkInterest = LaunchConfigDescription.World.LegacyFlags.Find(TEXT("enable_chunk_interest")))
-	{
-		if (SpatialGDKRuntimeSettings->bUsingQBI && (*EnableChunkInterest == TEXT("true")))
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: What about for one version we keep a variant of this check in that validates that you *don't* have enable_chunk_interest set to true in your launch config? If you do, print a message about removing it and return false.

**Pull request merged**